composer/callbacks/system_metrics_monitor.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""System metrics monitor callback."""
+
+from __future__ import annotations
+
+import logging
+import os
+
+import psutil
+
+from composer.core import Callback, Event, State
+from composer.loggers import Logger
+from composer.utils import dist
+from composer.utils.import_helpers import MissingConditionalImportError
+
+log = logging.getLogger(__name__)
+
+__all__ = ['SystemMetricsMonitor']
+
+
+class SystemMetricsMonitor(Callback):
+    """Track system metrics."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        try:
+            import pynvml
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='pynvml',
+                                                conda_package='pynvml',
+                                                conda_channel='conda-forge') from e
+        pynvml.nvmlInit()
+
+    def run_event(self, event: Event, state: State, logger: Logger):
+        # skip the microbatch events
+        if event in [
+                Event.BEFORE_FORWARD,
+                Event.AFTER_FORWARD,
+                Event.BEFORE_LOSS,
+                Event.AFTER_LOSS,
+                Event.BEFORE_BACKWARD,
+                Event.AFTER_BACKWARD,
+        ]:
+            return
```

Comment:
Instead of this, can we be explicit in which events to run on? Maybe `BATCH_START`/`EVAL_BATCH_START`/`PREDICT_BATCH_START` is sufficient? There's many events that run very fast otherwise, eg
```
self.engine.run_event(Event.AFTER_DATALOADER)

self.engine.run_event(Event.BATCH_START)
```
so this might log too much?